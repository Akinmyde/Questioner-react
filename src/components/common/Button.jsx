import React from 'react';
import propTypes from 'prop-types';

const Button = ({ value, ...rest }) => (<button {...rest} type="button">{value}</button>);

Button.defaultProps = {
  className: propTypes.defaultProps = 'btn',
};

Button.propTypes = {
  id: propTypes.string.isRequired,
  value: propTypes.string.isRequired,
  onClick: propTypes.func.isRequired,
  className: propTypes.string,
};

export default Button;
