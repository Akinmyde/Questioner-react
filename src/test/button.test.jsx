import 'babel-polyfill';
import React from 'react';
import { shallow } from 'enzyme';
import Button from '../components/common/Button';

describe('Button Test', () => {
  it('should render button correctly', () => {
    const button = shallow(<Button id="test" value="text" onClick={() => {console.log('this is clicked')}} />);
    expect(button).toMatchSnapshot();
    expect(button.hasClass('btn'))
    expect(button.text='text')
  });
});
