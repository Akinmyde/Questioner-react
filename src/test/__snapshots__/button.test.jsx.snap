// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Test should render button correctly 1`] = `
<button
  className="btn"
  id="test"
  onClick={[Function]}
  type="button"
>
  text
</button>
`;
